sdk/src/main/java/com/mapbox/maps/renderer/MapboxRenderThread.kt

```diff
@@ -307,6 +307,7 @@ internal class MapboxRenderThread : Choreographer.FrameCallback {
     renderDestroyCallChain = true
     mapboxRenderer.destroyRenderer()
     renderDestroyCallChain = false
+    nonRenderEventQueue.clear()
```

Comment:
Still concerned that `renderHandlerThread.clearMessageQueue()` is not executed here. That actually means that if we're using texture mode in current implementation and we do have some __non `EventType.DESTROY_RENDERER` SDK events__ in handler thread message queue - they will be actually executed with new surface (when `renderThreadPrepared` will become `true`) and new native renderer instance leading potentially to hard catching errors.

IMO we need to revert flag and call `renderHandlerThread.clearMessageQueue(clearSdkEventsOnly = true)` - in that case we will leave `DESTROY_RENDERER` in the queue and they will be executed ASAP (as they don't need `renderThreadPrepared == true`) but truly render events will be cleared. However ideally we should not clear user events then meaning we still need to differentiate SDK and non-SDK events.

Reply:
Added clearing messages in the texture view destroy usecase as well. However I dont think we need to keep user's render events in that case, since: 
- it's an edge case for the old Androids, so will rarely happen
- in theory events may be triggered by map loaded events, so will depend on the renderer lifecycle, thus executing the events is probably not expected. Lets sync offline on that